packages/kbn-dev-utils/src/kbn_client/kbn_client_requester.ts

```diff
@@ -70,20 +69,40 @@ const delay = (ms: number) =>
     setTimeout(resolve, ms);
   });
 
+export interface KibanaConfig {
+  url: string;
+  ssl:
+    | undefined
+    | {
+        enabled: boolean;
+        key: string;
+        certificate: string;
+        certificateAuthorities: string;
+      };
+}
+
 export class KbnClientRequester {
-  constructor(private readonly log: ToolingLog, private readonly kibanaUrls: string[]) {}
+  private readonly httpsAgent: Https.Agent | null;
+  constructor(private readonly log: ToolingLog, private readonly kibanaConfig: KibanaConfig) {
+    this.httpsAgent =
+      kibanaConfig.ssl && kibanaConfig.ssl.enabled
+        ? new Https.Agent({
+            cert: kibanaConfig.ssl.certificate,
+            key: kibanaConfig.ssl.key,
+            ca: kibanaConfig.ssl.certificateAuthorities,
+          })
+        : null;
+  }
 
   private pickUrl() {
-    const url = this.kibanaUrls.shift()!;
-    this.kibanaUrls.push(url);
-    return url;
+    return this.kibanaConfig.url;
   }
 
   public resolveUrl(relativeUrl: string = '/') {
     return Url.resolve(this.pickUrl(), relativeUrl);
   }
 
-  async request<T>(options: ReqOptions): Promise<T> {
+  async request<T>(options: ReqOptions): Promise<AxiosResponse<T>> {
```

Comment:
It's necessary for the Security plugin changes below. I could add something an optional `asResponse` flag instead, but I decided on KISS.